<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="input_detected_objects" default="/detection/lidar_detector/objects"/>
  <arg name="input_obstacle_pointcloud" default="/kitti/velo/pointcloud"/>
  <arg name="output_objects" default="/detection/lidar_filtered/objects"/>
  <arg name="obstacle_pointcloud_based_validator_param_path" default="$(find obstacle_pointcloud_validator)/config/obstacle_pointcloud_based_validator.param.yaml"/>

  <node pkg="obstacle_pointcloud_validator" type="obstacle_pointcloud_validator" name="obstacle_pointcloud_validator" output="screen">
    <param name="input_objects_topic_" value="$(arg input_detected_objects)"/>
    <param name="obstacle_pointcloud_topic_" value="$(arg input_obstacle_pointcloud)"/>
    <param name="output_objects_topic_" value="$(arg output_objects)"/>
    <rosparam from="$(var obstacle_pointcloud_based_validator_param_path)"/>
  </node>

  <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="cluster_detect_visualization_02"
        output="screen" ns="/detection/lidar_filtered">
    <param name="objects_src_topic" value="/objects"/>
  </node>
</launch>
